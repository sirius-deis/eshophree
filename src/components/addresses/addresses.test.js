import { render, screen } from '@testing-library/react';
import Addresses from './addresses';

import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

describe('Address component', () => {
  let store;
  //id, title, name, address, city, state, zipCode, phone, email
  const address = {
        id: 1,
        title: 'Billing Address',
        name: 'John Doe',
        street: '123 Main St',
        city: 'Anytown',
        state: 'CA',
        zipCode: '12345',
        phone: '123-456-7890',
        email: 'john.doe@example.com'
      };

  beforeEach(() => {
    store = mockStore({user: {isLoading: false, error: null, addresses: [address]}});
    store.dispatch = jest.fn();
  });
  it('should match snapshot', () => {
    const { container } = render(<Provider store={store}><Addresses /></Provider>);
    expect(container).toMatchSnapshot();
  });
  it('should render the address component with default props', () => {
    render(<Provider store={store}><Addresses /></Provider>);
    expect(screen.getByText(address.title)).toBeInTheDocument();
    expect(screen.getByText(address.name)).toBeInTheDocument();
    expect(screen.getByText(address.street)).toBeInTheDocument();
    expect(screen.getByText(new RegExp(address.city))).toBeInTheDocument();
    expect(screen.getByText(new RegExp(address.state))).toBeInTheDocument();
    expect(screen.getByText(new RegExp(address.zipCode))).toBeInTheDocument();
    expect(screen.getByText(address.phone)).toBeInTheDocument();
    expect(screen.getByText(address.email)).toBeInTheDocument();
  });

  it('should handle missing address gracefully', () => {
    render(<Provider store={mockStore({user: {isLoading: false, error: null, addresses: []}})}><Addresses /></Provider>);
    expect(screen.getByText(/no address available/i)).toBeInTheDocument();
  });
});