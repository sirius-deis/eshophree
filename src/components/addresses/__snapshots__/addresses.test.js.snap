// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address component should match snapshot 1`] = `
<div>
  <div
    class="sc-bgqQcB cgjxUA"
  >
    <h2
      class="sc-gTRrQi"
    >
      The following addresses will be used on the checkout page by default
    </h2>
    <div
      class="sc-kcuKUB dTGKrC"
    >
      <div
        class="sc-ispOId dbDVvX"
      >
        <h2
          class="sc-dcrmVg kcsjkr"
        >
          Billing Address
        </h2>
        <button
          class="sc-fFGjHI sc-gxYJeL ivVKaH gIQAZC"
          type="button"
        >
          Edit
        </button>
      </div>
      <div
        class="sc-kPTPQs zAAxG"
      />
      <h3>
        John Doe
      </h3>
      <div
        class="sc-crozmw jOfitN"
      >
        <p
          class="sc-jJTsDX gAVfzf"
        >
          123 Main St
        </p>
        <p
          class="sc-jJTsDX gAVfzf"
        >
          Anytown
          , 
          CA
           
          12345
        </p>
        <p
          class="sc-jJTsDX gAVfzf"
        >
          123-456-7890
        </p>
        <p
          class="sc-jJTsDX gAVfzf"
        >
          john.doe@example.com
        </p>
      </div>
    </div>
  </div>
</div>
`;
